jbmc/src/java_bytecode/java_string_literals.h

```diff
@@ -12,9 +12,25 @@ Author: Chris Smowton, chris.smowton@diffblue.com
 #include <util/symbol_table.h>
 #include <util/std_expr.h>
 
+/// Creates or gets an existing constant global symbol for a given string
+/// literal.
+/// \param string_expr: string literal expression to convert
+/// \param symbol_table: global symbol table. If not already present, constant
+///   global symbols will be added.
+/// \param string_refinement_enabled: if true, string refinement's string data
+///   structure will also be initialised and added to the symbol table.
+/// \return a symbol_expr corresponding to the new or existing literal symbol.
 symbol_exprt get_or_create_string_literal_symbol(
   const exprt &string_expr,
```

Comment:
Are there any users of this overload left?

Reply:
Only one:
https://github.com/diffblue/cbmc/blob/7dd720b5db62d05bf814b4428c90dcca816d425a/jbmc/src/java_bytecode/java_bytecode_language.cpp#L424

Reply:
In that case it would be a very welcome addition to this PR to make `java_string_literal_exprt` rather than directly twiddling operands / named subexpressions -- fair enough though if you think that's out of scope.

Reply:
It looks like there aren't any `java[\w_]+exprt` types yet. Should such a type go into `jbmc/src/java_bytecode/java_types.h` or a new file, e.g. something like `jbmc/src/java_bytecode/java_expressions.h` or similar?
Or does it make more sense to create a new folder `jbmc/src/util`? In that case I would definitely make a new PR for that.

Reply:
Does symex have to handle these expressions in a specific way? If it's the case then they shouldn't be defined in a `jbmc` subdirectory. Otherwise `java_expr.h` would be clearer for showing the class inherits from `exprt` and I'd prefer one class per file, so `java_string_literal_expr.h` would be better.

Reply:
Hearing about language-specific expressions makes me panic a bit. You will need to make sure all of these are gone by the time the code leaves the Java front-end.

Reply:
They are; these are just descriptions of the exprts produced by the bytecode parser. The only other example I'm aware of now, and which I suggest we emulate, is `fieldref_exprt`, which is defined in `java_bytecode_parse_tree.h`.

Reply:
(that expression is also local to the parser and becomes `member_exprt` later once we have more information about what it's referring to)

Reply:
I added the new class as you suggested, in a new commit.